include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -2233,9 +2238,10 @@ TriaAccessor<0, dim, spacedim>::operator ++ () const
 template <int dim, int spacedim>
 inline
 void
-TriaAccessor<0, dim, spacedim>::operator -- () const
+TriaAccessor<0, dim, spacedim>::operator -- ()
 {
-  Assert (false, ExcNotImplemented());
+  if (global_vertex_index!=numbers::invalid_unsigned_int)
+    --global_vertex_index;
```

Comment:
That's exactly what I mean. It *happens* to be defined that way, but it's supposed to be used as an *exceptional* value outside the usual number system. We could have chosen it to be `(unsigned int)(-2)` as well and everything should continue to work. The fact that you get it by decrementing -1 is a coincidence.